perun-core/src/test/java/cz/metacentrum/perun/core/entry/MembersManagerEntryIntegrationTest.java

```diff
@@ -2207,6 +2208,70 @@ public void getAllSponsoredMembers() throws Exception {
 		assertTrue(allSponsoredMembers.contains(richMember2));
 	}
 
+	@Test
+	public void moveMembership() throws Exception {
+		System.out.println(CLASS_NAME + "moveMembership");
+
+		User sourceUser = perun.getUsersManagerBl().getUserByMember(sess, createdMember);
+
+		Member member = setUpMember(createdVo);
+		User targetUser = perun.getUsersManagerBl().getUserByMember(sess, member);
+		// delete member - member was created just to create target user
+		perun.getMembersManager().deleteMember(sess, member);
+		// save user for deletion after test
+		usersForDeletion.add(targetUser);
+
+		// add createdMember to createdGroup
+		groupsManagerEntry.addMember(sess, createdGroup, createdMember);
+
+		// set attributes
+		Attribute memberAttribute = setUpAttribute(Integer.class.getName(), "testMemberAttribute", AttributesManager.NS_MEMBER_ATTR_DEF, 15);
+		perun.getAttributesManagerBl().setAttribute(sess, createdMember, memberAttribute);
+		Attribute memberGroupAttribute = setUpAttribute(Integer.class.getName(), "testMemberGroupAttribute", AttributesManager.NS_MEMBER_GROUP_ATTR_DEF, 15);
+		perun.getAttributesManagerBl().setAttribute(sess, createdMember, createdGroup, memberGroupAttribute);
+
+		List<Attribute> sourceMemberAttrs = perun.getAttributesManagerBl().getAttributes(sess, createdMember);
+		List<Attribute> sourceMemberGroupAttrs = perun.getAttributesManagerBl().getAttributes(sess, createdMember, createdGroup);
+
+		BanOnVo banOnVo = new BanOnVo(-1, createdMember.getId(), createdMember.getVoId(), null, "ban");
+		perun.getVosManagerBl().setBan(sess, banOnVo);
+
+		perun.getMembersManager().moveMembership(sess, createdVo, sourceUser, targetUser);
+
+		Member targetMember = perun.getMembersManagerBl().getMemberByUser(sess, createdVo, targetUser);
+		List<Attribute> targetMemberAttrs = perun.getAttributesManagerBl().getAttributes(sess, targetMember);
+		List<Attribute> targetMemberGroupAttrs = perun.getAttributesManagerBl().getAttributes(sess, targetMember, createdGroup);
+
+		assertEquals("The source and target member attributes should have the same size.",
```

Comment:
Nit: using assertJ, you can simplify these asserts e.g. like this:
`assertThat(fooA).containsExactlyInAnyOrderElementsOf(fooB);`

AssertJ provides a lot of usefull assertions, that make tests easier to read and also write.